ctapipe/image/muon/muon_ring_finder.py

```diff
@@ -69,3 +170,16 @@ def fit(self, x: u.deg, y: u.deg, weight, times=None):
         output.ring_fit_method = "ChaudhuriKundu"
 
         return output
+
+    def fit(self, x, y, img):
+        """allows any fit to be called in form of
+            MuonRingFitter(fit_method = "name of the fit")
+        """
+        self.x = x
```

Comment:
There is no reason, x and y should be stored to the instance here. It's not necessary to add this statefulness.

Just hand it over to the fit methods, which then can be free functions.

Reply:
fixed that